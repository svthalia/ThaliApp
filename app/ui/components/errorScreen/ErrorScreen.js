import React from 'react';
import { Image, Text, View } from 'react-native';
import { withTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import styles from './style/ErrorScreen';

const cloud = require('../../../assets/img/sad_cloud.png');

const ErrorScreen = (props) => (
  <View
    style={styles.content}
  >
    <Image
      source={cloud}
      style={styles.image}
    />
    <Text style={styles.text}>
      {props.message}
    </Text>
    <Text style={styles.text}>
      {props.t('Try again later.')}
    </Text>
  </View>
);

ErrorScreen.propTypes = {
  message: PropTypes.string.isRequired,
  t: PropTypes.func.isRequired,
};

export default withTranslation('ui/components/errorScreen/ErrorScreen')(ErrorScreen);
