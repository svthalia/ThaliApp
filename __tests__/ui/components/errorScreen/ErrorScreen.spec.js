import React from 'react';
import renderer from 'react-test-renderer';
import ErrorScreen from '../../../../app/ui/components/errorScreen/ErrorScreen';

describe('ErrorScreen component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ErrorScreen message='Lorem ipsum' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
