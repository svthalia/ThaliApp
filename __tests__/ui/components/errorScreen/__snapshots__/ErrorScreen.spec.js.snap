// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source="test-file-stub"
    style={
      Object {
        "height": 200,
        "marginBottom": 32,
        "resizeMode": "contain",
        "width": 200,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#616161",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    Lorem ipsum
  </Text>
  <Text
    style={
      Object {
        "color": "#616161",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    Try again later.
  </Text>
</View>
`;
